UnitTest/tests/test_DeviceData_rmsg-definition.txt

```diff
@@ -61,18 +61,23 @@ defmod test_DeviceData_rmsg UnitTest dummyDuino (
 				$dmsg_matched=0;
 				$SD_Dispatch_calledCounter=0;
 	            SIGNALduino_Parse($targetHash, $targetHash, $targetHash->{NAME}, "\002".$tData->{rmsg}."\003") if (defined($tData->{rmsg}));
-	            if ($SD_Dispatch->called() >0 )  {
-					$noTestRun=0;
-					my $expected_repeat=min(4,$tData->{repeat}+1);
-					 
-					TODO: {
-					    local $TODO = "Checking for repeats is not finishes, because number of repeats is not always number of dispatches";
-				        is($SD_Dispatch_calledCounter, $expected_repeat, "check number of dispatch attempts") if (defined($tData->{repeat}));
-					};
-					
-		            ok($id_matched,"SIGNALduno_Dispatch check id ") || note explain (($SD_Dispatch->called_with())[4] , " vs ", $testSet->{id});
-		            ok($dmsg_matched,"SIGNALduno_Dispatch check dmsg ") || note explain (($SD_Dispatch->called_with())[2] , " vs ", $testSet->{dmsg});
-		      		            
+            if ($SD_Dispatch->called() >0 )  {
+              $noTestRun=0;
+
+							ok($id_matched,"SIGNALduno_Dispatch check id ") || note explain (($SD_Dispatch->called_with())[4] , " vs ", $testSet->{id});
+              ok($dmsg_matched,"SIGNALduno_Dispatch check dmsg ") || note explain (($SD_Dispatch->called_with())[2] , " vs ", $testSet->{dmsg});
+
+              if (defined($tData->{dispatch_repeats})) {
+								ok( $tData->{dispatch_repeats} , "JSON dispatch_repeats entry available");
+								## some tests please added in separate Travistest on SIGNALduino_TOOL ##
+								ok( $tData->{dispatch_repeats} =~ /\d+/ , "JSON dispatch_repeats entry check numeric");
+							}
+
+							my $expected_repeat=min(1,$tData->{dispatch_repeats});
```

Comment:
wieso hast Du hier min auf 1 geändert?

Reply:
Es gibt doch mindestens immer 1 oder habe ich da falsche Denkweisen?
Das Ergebnis auf dem lokalen System war für mich plausible hm..

Reply:
Die 4, kam von dem Default von maxMuMsgRepeat, was auf 4 steht.

In dem Fall wäre expected_repeat eigentlich immer 1 oder hab ich was übersehen? 
Korrekter wäre es natürlich das Attribut abzufragen und nicht einfach eine 4 schreiben und keiner kann ahnen wie ich auf die gekommen bin :)

Reply:
> Korrekter wäre es natürlich das Attribut abzufragen und nicht einfach eine 4 schreiben und keiner kann ahnen wie ich auf die gekommen bin :)

Dann sollten wir es mal versuchen einzurichten ;-)

Reply:
Willst Du oder soll ich?

Ich nehme an ein `AttrVal($target,"maxMuMsgRepeat",4) `sollte passen